src/System.Private.ServiceModel/tools/test/Bridge/Bridge/Program.cs

```diff
@@ -18,10 +21,173 @@ private static void Main(string[] args)
         {
             CommandLineArguments commandLineArgs = new CommandLineArguments(args);
 
+            Console.WriteLine("Specified BridgeConfiguration is:{0}{1}",
+                                Environment.NewLine, commandLineArgs.BridgeConfiguration.ToString());
+
+            // If asked to ping (not the default), just ping and return an exit code indicating its state
+            if (commandLineArgs.Ping)
+            {
+                string pingResult = PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
+                               commandLineArgs.BridgeConfiguration.BridgePort);
+                if (pingResult == null)
+                {
+                    Console.WriteLine("The Bridge is running.");
+                    Environment.Exit(0);
+                }
+                else
+                {
+                    Console.WriteLine("The Bridge is not running: {0}", pingResult);
+                    Environment.Exit(1);
+                }
+            }
+            else if (commandLineArgs.StopIfLocal)
+            {
+                StopBridgeIfLocal(commandLineArgs);
+            }
+            else if (commandLineArgs.Stop)
+            {
+                StopBridge(commandLineArgs);
+            }
+            else
+            {
+                // Default action is starting the Bridge
+                StartBridge(commandLineArgs);
+            }
+        }
+
+        private static string PingBridge(string host, int port)
```

Comment:
This method probably needs a comment... It's not intuitive that when PingBridge returns `null`, it's a success case (Bridge is up), and when PingBridge returns a non-`null`, it's an error case (Bridge is down). Took me a couple of reads to realize that :)